core/include/cubos/core/reflection/traits/inherits.hpp

```diff
@@ -0,0 +1,41 @@
+#include <string>
+#include <unordered_set>
+
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::core::reflection
+{
+    /// @brief Provides inheritance relationship between types.
+    /// @see See @ref examples-core-reflection-traits-inherits for an example of using this trait.
+    /// @ingroup core-reflection
+    class InheritsTrait
+    {
+    public:
+        ~InheritsTrait() = default;
+
+        /// @brief Constructs.
+        InheritsTrait() = default;
+
+        /// @brief Adds an inheritance relationship to the type.
+        /// @tparam T The parent type from which this type inherits.
+        /// @return Trait.
+        template <typename T>
+        InheritsTrait&& from() &&
```

Comment:
I would change this to create a trait, instead of changing the existing one.
```suggestion
        static InheritsTrait from()
```